cmd/bucket-replication.go

```diff
@@ -193,7 +209,7 @@ func replicateObject(ctx context.Context, bucket, object, versionID string, obje
 	putOpts := putReplicationOpts(dest, objInfo)
 
 	replicationStatus := replication.Complete
-	_, err = tgt.PutObject(ctx, dest.Bucket, object, gr, size, "", "", putOpts)
+	_, err = tgt.PutObject(context.Background(), dest.Bucket, object, gr, size, "", "", putOpts)
```

Comment:
with original context, replication fails even though caller is not canceling context. Leaving this here for now

Reply:
we need to see why it fails @poornas removing that means there is no way to cancel this call.

Reply:
It is failing because mc now automatically cancels context after call completes - replication runs after the Put completes, so it needs its own context. Doing this in object-handlers while calling replicateObject